config/destination_test.go

```diff
@@ -0,0 +1,153 @@
+// limitations under the License.
+
+package config
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/config/validator"
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/models"
+)
+
+func TestParseDestination(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+		in   map[string]string
+		want Destination
+		err  error
+	}{
+		{
+			name: "valid config with only required fields",
+			in: map[string]string{
+				models.ConfigPrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+				models.ConfigClientEmail: "test@test-pubsub.com",
+				models.ConfigProjectID:   "test-pubsub",
+				models.ConfigTopicID:     "test-TOPic.scription~%_1230+",
+			},
+			want: Destination{
+				General: General{
+					PrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+					ClientEmail: "test@test-pubsub.com",
+					ProjectID:   "test-pubsub",
+				},
+				TopicID: "test-TOPic.scription~%_1230+",
+			},
+		},
+		{
+			name: "topic id is too small",
+			in: map[string]string{
+				models.ConfigPrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+				models.ConfigClientEmail: "test@test-pubsub.com",
+				models.ConfigProjectID:   "test-pubsub",
+				models.ConfigTopicID:     "to",
+			},
+			err: validator.InvalidNameErr(models.ConfigTopicID),
+		},
+		{
+			name: "topic id is too big",
+			in: map[string]string{
+				models.ConfigPrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+				models.ConfigClientEmail: "test@test-pubsub.com",
+				models.ConfigProjectID:   "test-pubsub",
+				models.ConfigTopicID: "test-KfAvXlueAOAL5UhJ1Rr4zzsZJnB1ilOYXrxgjUmJqD1TZdEC8NAl22oPrzx1HpB7MfgteQUMn8u" +
+					"NCadTgrbUJKY6Lb6ARzyOY3bI3W6YjadDLTl47DIqA7zjYYQNIud9PHXgA0v3NVlk2AVLaziUwylawemiUJOee68ULPg" +
+					"GyBgoCIMAB7ukAgjN0fhGPeYART2yojioOp3w9mBPdklk7OY8rJQFCy5ii70byyHFqT3JG00kJTPzdPPdt53",
+			},
+			err: validator.InvalidNameErr(models.ConfigTopicID),
+		},
+		{
+			name: "topic id does not start with a letter",
+			in: map[string]string{
+				models.ConfigPrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+				models.ConfigClientEmail: "test@test-pubsub.com",
+				models.ConfigProjectID:   "test-pubsub",
+				models.ConfigTopicID:     "1-test-top*",
+			},
+			err: validator.InvalidNameErr(models.ConfigTopicID),
+		},
+		{
+			name: "topic id has unsupported characters",
+			in: map[string]string{
+				models.ConfigPrivateKey:  "-----BEGIN PRIVATE KEY-----\nMIIEvAIBADAQEFAASC-----END PRIVATE KEY-----",
+				models.ConfigClientEmail: "test@test-pubsub.com",
+				models.ConfigProjectID:   "test-pubsub",
+				models.ConfigTopicID:     "test-top*",
+			},
+			err: validator.InvalidNameErr(models.ConfigTopicID),
+		},
+		{
+			name: "topic id starts with goog",
```

Comment:
This is not supported? I didn't know that.

Reply:
Correct.
https://cloud.google.com/pubsub/docs/admin#resource_names